storage-operator/pkg/salt/client.go

```diff
@@ -232,25 +205,73 @@ func (self *Client) GetVolumeSize(
 	}
 
 	self.logger.Info(
-		"GetVolumeSize", "Volume.NodeName", nodeName, "Volume.Name", volumeName,
+		jobName, "Volume.NodeName", nodeName, "Volume.Name", volumeName,
+	)
+
+	return self.submitJob(ctx, jobName, volumeName, payload)
+}
+
+// Return the name of the block device designed by `devicePath`.
+//
+// This request is synchronous.
```

Comment:
Back to using synchronous requests against SaltAPI? Isn't it a recipe for failure?

Reply:
As I tried to explain in the commit message, I think here it's acceptable because unlike `disk.dump` (which is in the "critical path" of volume creation):

- it doesn't prevent the volume to be used: a failed `disk.dump` can block/delay the creation of the volume, where a failed `metalk8s_volume.device_name` won't change the status/avaibility of the volume (volume is already available and will stay that way, worst case is we'll miss some state in the UI until the call succeed, but this doesn't prevent the volume to be ready and used).
- it's always retried on failure, no matter the "type" of error: unlike `disk.dump` (which only retry on some error and "abort" on some others), `metalk8s_volumes.device_name` is always retried on error

Given those difference, I think we can use a sync call (and have a big code simplification), without the drawbacks we used to have with `disk.dump`.

Reply:
Sorry I missed the commit message, OK for me!